### ISSUE TITLE
Add system for loading parameters from a python file

### ISSUE THREAD

**Issue by JeremyMcCormick** [JeffersonLab/hps-mc#62]
Instead of complex command line argument interfaces, it might be useful to be able to load a python file with job parameters, e.g.

```
apmass = 40.0
nevents = 10000
param_key = "1pt05"
rand_seed = 1
ctau = 1.0
zvert = -5.0
filename = "ap"
cond_detector = "HPS-EngRun2015-Nominal-v5-0-fieldmap"
```

Probably calling `execfile()` would do this job well enough.

Additional ideas here:

https://stackoverflow.com/questions/924700/best-way-to-retrieve-variable-values-from-a-text-file-python-json

**Comment by mholtrop** [comment]
If you are going to do that, then at least use JSON: https://docs.python.org/3/library/json.html
In other words:
apmass: 40.0
nevents: 10000

**Comment by JeremyMcCormick** [comment]
Why wouldn't we just make it a python file rather than JSON?  Is there some other system you want to load this data into?

**Comment by JeremyMcCormick** [comment]
I guess JSON would actually be something like:

```
{
    "apmass": 40.0,
    "nevents":10000
}
```

etc.

**Comment by mholtrop** [comment]
OK, that works too. I thought you were going to parse the text file. You are planning to import this?

**Comment by JeremyMcCormick** [comment]
Actually, now that I think about it JSON would be nice here as it is portable.

So I can do something like this for the basic class:

```
class JobParameters:
    
    def __init__(self, filename):
        self.load(filename)
    
    def load(self, filename):
        rawdata = open(filename, 'r').read()
        self.json_dict = json.loads(rawdata)

    def __getattr__(self, name):
        if name in self.json_dict:
            return self.json_dict[name]
        else:
            return self.__getattribute__(name)
```

The access in the job by attributes on that class like:

```
params = JobParameters(sys.argv[1])
apmass = params.apmass
nevents = params.nevents
```

It's a nice solution I think.

I want to do this because if you work from command line arguments it gets kind of crazy (each type of MC job then ends up having an extremely complex command line interface).

**Comment by omar-moreno** [comment]
If you are considering JSON, why not use YAML instead:

http://yaml.org/

**Comment by JeremyMcCormick** [comment]
> why not use YAML instead

Let's not overcomplicate things.  ;-)

**Comment by omar-moreno** [comment]
YAML is simpler to use, it doesn't require you to add another class like
you are planning to (single line of code that loads a file and returns a
dictionary) and it allows you to comment the configuration files. Just
trying to make your life easier.

**Comment by JeremyMcCormick** [comment]
Are you saying this isn't completely self-explanatory?  :)

```
{
    "apmass": 40.0,
    "nevents": 500,
    "run_params": "1pt05",
    "seed": 1,
    "ctau": 1.0,
    "z": -5.0,
    "filename": "ap",
    "detector": "HPS-EngRun2015-Nominal-v5-0-fieldmap",
    "run": 5772,
    "readout_steering": "/org/hps/steering/readout/EngineeringRun2015TrigPairs1_Pass2.lcsim",
    "recon_steering": "/org/hps/steering/recon/EngineeringRun2015FullReconMC.lcsim"
}
```

**Comment by JeremyMcCormick** [comment]
Resolved by PR #66.
